packages/kbn-es/src/cluster.js

```diff
@@ -274,12 +274,14 @@ exports.Cluster = class Cluster {
       (options.esEnvVars.ES_JAVA_OPTS ? `${options.esEnvVars.ES_JAVA_OPTS} ` : '') +
       '-Xms1g -Xmx1g';
 
+    // By default, we want to always unset JAVA_HOME so that the bundled JDK will be used
+    options.esEnvVars.JAVA_HOME = options.esEnvVars.JAVA_HOME || '';
```

Comment:
nit: Could put `JAVA_HOME` directly into the `env` object instead of mutating `options.esEnvVars`. If placed where it previously was, it would also allow for it to be overwritten if needed.